stdlib/public/runtime/Errors.cpp

```diff
@@ -188,11 +194,16 @@ reportOnCrash(uint32_t flags, const char *message)
 static void
 reportNow(uint32_t flags, const char *message)
 {
+#if defined(_MSC_VER)
+#define STDERR_FILENO 2
+  _write(STDERR_FILENO, message, strlen(message));
+#else
   write(STDERR_FILENO, message, strlen(message));
+#endif
 #ifdef __APPLE__
   asl_log(NULL, NULL, ASL_LEVEL_ERR, "%s", message);
 #endif
-#if !defined(__CYGWIN__) && !defined(__ANDROID__)
+#if !defined(__CYGWIN__) && !defined(__ANDROID__) && !defined(_MSC_VER)
```

Comment:
This condition (`!defined(__CYGWIN__) && !defined(__ANDROID__) && !defined(_MSC_VER)`) is becoming more complex and it is repeated in a few places in this file.  Could you factor it into a macro?  Something like

```c
#if !defined(__CYGWIN__) && !defined(__ANDROID__) && !defined(_MSC_VER)
#  define SWIFT_SUPPORTS_BACKTRACE_REPORTING 0
#else
#  define SWIFT_SUPPORTS_BACKTRACE_REPORTING 1
#endif
```